File: week4_tutorial/threads/threads.py
import threading
global arr
arr = [0,0,2,0,0,0,0,0,0,0]

def print_numbers():
    for i in range(1, 6):
        print("Number:", i)

def print_letters():
    for letter in 'abcde':
        print("Letter:", letter)

# Create two threads
thread1 = threading.Thread(target=print_numbers)
thread2 = threading.Thread(target=print_letters)

# Start the threads
thread1.start()
thread2.start()

# Wait for both threads to finish
thread1.join()
thread2.join()

print("Both threads have finished")

File: week4_tutorial/threads/threads.cs
using System;
using System.Threading;

class Program
{
    static void PrintNumbers()
    {
        for (int i = 1; i <= 5; i++)
        {
            Console.WriteLine("Number: " + i);
        }
    }

    static void PrintLetters()
    {
        foreach (char letter in "abcde")
        {
            Console.WriteLine("Letter: " + letter);
        }
    }

    static void Main()
    {
        Thread thread1 = new Thread(PrintNumbers);
        Thread thread2 = new Thread(PrintLetters);

        thread1.Start();
        thread2.Start();

        thread1.Join();
        thread2.Join();

        Console.WriteLine("Both threads have finished");
    }
}
